### ISSUE TITLE
Snackbar - won't show notification if "goto" away and "goto" back

### ISSUE THREAD

**Issue by kuchaguangjie** [hperrin/svelte-material-ui#495]
**Describe the bug**
There is a component contains a `<Snackbar bind:this={notifyBar}>`, it can show notification.

But, if I call `goto()` to go away, then call `goto()` to come back, then it won't show the notification any more, need a refresh to work again.

I've check that the  binded `notifyBar` is not null, it's still there. It just won't show anything.
And there is no error in the chrome devtool's console.

What happened when call `goto()`, is there anyway to re-init the notifier without refresh?
Any idea?

**Desktop (please complete the following information):**
 - OS: linux mint mate 20
 - Browser: chrome
 - Version: 102.0.5005.115 (Official Build) (64-bit)

**Comment by kuchaguangjie** [comment]
Stackoverflow question: https://stackoverflow.com/questions/73216615/

**Comment by NickantX** [comment]
Hello. Since you're using SvelteKit, you may want to consider an alternative solution such as moving your Snackbar into your `__layout.svelte`, and exposing a function to show this notification throughout your app by passing a function with [setContext](https://svelte.dev/docs#run-time-svelte-setcontext). To make this whole process more convenient I would recommend using SMUI's "Kitchen Snackbar generator" as shown at the bottom of the [documentation for Snackbars](https://sveltematerialui.com/demo/snackbar/).
This solution will not only make it super convenient to push notifications to your Snackbar from anywhere in the app (simple call to getContext to get the function), it will also ensure the Snackbar doesn't get unmounted/remounted during navigation and should fix your issue.

**Comment by kuchaguangjie** [comment]
@NickantX I've tried the `setContext()` solution, it works, a small problem is after jump to another route the notification remains, I have to clear it by hand for each page.

And, didn't try the `Kitchen Snackbar` yet, seems not relevant to this issue directly, not sure.

**Comment by NickantX** [comment]
> a small problem is after jump to another route the notification remains, I have to clear it by hand for each page

My bad, I didn't realize you wanted the Snackbar to dismiss after navigating to another page. Since the Snackbar is now in your global layout component, in your layout you can import `afterNavigate` from `$app/navigation`, and use this to close the Snackbar.

> And, didn't try the Kitchen Snackbar yet, the doc is very simple, seems not relevant to this issue directly, not sure.

Kitchens are essentially a convenient API that generate Snackbars (and queues them!) for you with a simple JavaScript function call, you can use it in pretty much any place your app may need a Snackbar.

**Comment by kuchaguangjie** [comment]
Seems adding `afterNavigate()` to __layout.svelte, will cause every navigation a page refresh.

**Comment by NickantX** [comment]
> Seems adding `afterNavigate()` to __layout.svelte, will cause every navigation a page refresh.

afterNavigate is a lifecycle function, similar to onMount or onDestroy. pass a function to it that will close your snackbar.

**Comment by kuchaguangjie** [comment]
Yes, but as long as I add `afterNavigate()` all navigation become refresh ...

**Comment by hperrin** [comment]
The `Kitchen` component should be in your top level route, and shouldn't be unmounted. It's not designed to work that way.

You can pass it down to lower components with a writable store in a svelte context.
